ports/stm32/rfcore.c

```diff
@@ -261,9 +264,7 @@ void ipcc_init(uint32_t irq_pri) {
     // Enable receive IRQ on the BLE channel.
     LL_C1_IPCC_EnableIT_RXO(IPCC);
     LL_C1_IPCC_DisableReceiveChannel(IPCC, LL_IPCC_CHANNEL_1 | LL_IPCC_CHANNEL_2 | LL_IPCC_CHANNEL_3 | LL_IPCC_CHANNEL_4 | LL_IPCC_CHANNEL_5 | LL_IPCC_CHANNEL_6);
-    #if !MICROPY_HW_STM32WB_TRANSPARENT_MODE
     LL_C1_IPCC_EnableReceiveChannel(IPCC, IPCC_CH_BLE);
```

Comment:
This needs to be only enabled if NimBLE is being used. I think probably the simplest way to do this is to move this call to the start of `rfcore_ble_init` (at which point you'll be able to check the state of `mp_bluetooth_nimble_ble_state`).

I think down the track I'd probably make it so the caller of `rfcore_ble_init` (whcih is currently either transparent mode or hci uart init) pass a `use_irqs` bool or something, but what you have seems fine for now.

It would be worth adding a check to stm.rfcore_transparent() to early-fail if NimBLE is started too.

Reply:
I was originally going to add an arg to `rfcore_ble_init` for this however it'd still need a (new?) global variable to know whether to allow/block the irq re-enable in: 
```
STATIC void tl_check_msg(volatile tl_list_node_t *head, unsigned int ch, parse_hci_info_t *parse) {
        ...
        if (ch == IPCC_CH_BLE && (mp_bluetooth_nimble_ble_state != MP_BLUETOOTH_NIMBLE_BLE_STATE_OFF)) {
            // Renable IRQs for BLE now that we've cleared the flag.
            LL_C1_IPCC_EnableReceiveChannel(IPCC, IPCC_CH_BLE);
        }
}
```
as such re-using `mp_bluetooth_nimble_ble_state` is an efficient one to use, rather than creating something new.

Reply:
I've moved `LL_C1_IPCC_EnableReceiveChannel(IPCC, IPCC_CH_BLE);` as suggested in latest push